ratbagd/ratbagd-device.c

```diff
@@ -241,27 +241,35 @@ ratbagd_device_get_capabilities(sd_bus *bus,
 	struct ratbag_device *lib_device = device->lib_device;
 	enum ratbag_device_capability cap;
 	enum ratbag_device_capability caps[] = {
-		RATBAG_DEVICE_CAP_QUERY_CONFIGURATION,
+		RATBAG_DEVICE_CAP_CONFIGURATION,
 		RATBAG_DEVICE_CAP_RESOLUTION,
-		RATBAG_DEVICE_CAP_SWITCHABLE_RESOLUTION,
 		RATBAG_DEVICE_CAP_PROFILE,
-		RATBAG_DEVICE_CAP_SWITCHABLE_PROFILE,
-		RATBAG_DEVICE_CAP_DISABLE_PROFILE,
-		RATBAG_DEVICE_CAP_DEFAULT_PROFILE,
+		RATBAG_DEVICE_CAP_PROFILE_DISABLE,
+		RATBAG_DEVICE_CAP_PROFILE_DEFAULT,
 		RATBAG_DEVICE_CAP_BUTTON,
 		RATBAG_DEVICE_CAP_BUTTON_KEY,
 		RATBAG_DEVICE_CAP_BUTTON_MACROS,
 		RATBAG_DEVICE_CAP_LED,
 	};
 	size_t i;
 
-	CHECK_CALL(sd_bus_message_open_container(reply, 'a', "u"));
+	CHECK_CALL(sd_bus_message_open_container(reply, 'a', "(us)"));
 
 	for (i = 0; i < ELEMENTSOF(caps); i++) {
+		char *string;
+
 		cap = caps[i];
-		if (ratbag_device_has_capability(lib_device, cap)) {
-			CHECK_CALL(sd_bus_message_append(reply, "u", cap));
-		}
+		if (ratbag_device_has_capability(lib_device, cap, 'r') &&
```

Comment:
Not sure if this is worth it, but imo this would look nicer if you can have just `ratbag_device_has_capability(lib_device, cap, 'rw')`. The same goes for `ratbag_device_set_capability` and its `unset` inverse.

Reply:
true, but it makes things slightly harder to check too because you now have to loop over the string. Maybe as future patch?